src/lib/userStorage/UserProfileStorage.js

```diff
@@ -0,0 +1,456 @@
+  profileDefaults: {} = {
+    mobile: '',
+  }
+
+  profileSettings: {} = {
+    fullName: { defaultPrivacy: 'public' },
+    email: { defaultPrivacy: 'private' },
+    mobile: { defaultPrivacy: 'private' },
+    mnemonic: { defaultPrivacy: 'private' },
+    avatar: { defaultPrivacy: 'public' },
+    smallAvatar: { defaultPrivacy: 'public' },
+    walletAddress: { defaultPrivacy: 'public' },
+    username: { defaultPrivacy: 'public' },
+  }
+
+  subscribersProfileUpdates = []
+
+  walletAddressIndex = {}
+
+  //unecrypted profile field values
+  profile: { [key: string]: ProfileField } = {}
+
+  constructor(wallet: GoodWallet, profiledb: ProfileDB) {
+    // const seed = Uint8Array.from(Buffer.from(pkeySeed, 'hex'))
+    // this.privateKey = TextileCrypto.PrivateKey.fromRawEd25519Seed(seed)
+    this.wallet = wallet
+    this.profiledb = profiledb
+  }
+
+  /**
+   * read raw profile from database
+   */
+  async init() {
+    const rawProfile = await this.profiledb.getProfile()
+    this._setLocalProfile(await this._decryptProfileFields(rawProfile))
+  }
+
+  /**
+   * helper for setting value of local profile state
+   * @param newValue
+   * @private
+   */
+  _setLocalProfile(newValue) {
+    // this.subscribeProfileUpdates(newValue)
+    this.profile = newValue
+  }
+
+  /**
+   * helper for decrypt profile values
+   * @param profile
+   * @returns {Promise<{}>}
+   * @private
+   */
+  async _decryptProfileFields(profile): Promise<any> {
+    const outputProfile = {}
+    await Promise.all(
+      Object.keys(profile).map(
+        async item =>
+          typeof profile[item]?.value === 'string' &&
+          (outputProfile[item] = {
+            ...profile[item],
+            value: await this.profiledb.decryptField(profile[item]?.value),
+          }),
+      ),
+    )
+    return outputProfile
+  }
+
+  /**
+   * encrypt decrypted profile
+   * @param {*} profile
+   */
+  async _encryptProfileFields(profile): Promise<any> {
+    let encryptProfile = {}
+    await Promise.all(
+      Object.keys(profile).map(
+        async field =>
+          typeof profile[field]?.value === 'string' &&
+          (encryptProfile[field] = {
+            ...profile[field],
+            value: await this.profiledb.encryptField(profile[field]?.value),
+          }),
+      ),
+    )
+    return encryptProfile
+  }
+
+  /**
+   * saves a complete profile to the underlying storage
+   * @param {*} profile
+   * @param update
+   */
+  async setProfile(profile, update: boolean = false): Promise<any> {
+    if (update) {
+      const { getErrors, isValid, validate, ...profileFields } = profile
+      await Promise.all(
+        // eslint-disable-next-line no-return-await
+        Object.keys(profileFields).map(async key => await this.setProfileField(key, profileFields[key])),
+      )
+    } else {
+      const encryptedProfile = await this._encryptProfileFields(profile)
+      await this.profiledb.setProfile(encryptedProfile)
```

Comment:
instead of making single call to update the profile you make a call for each field and then another call to setProfile.
what you want is a helper that would modify the display value of each field in profile (just like setProfileField does)
then you encrypt the fields
then you save the whole profile to the database